src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx

```diff
@@ -180,11 +180,16 @@ function NodeComponent({
     };
 
     const makeNodeTree = (nodes: Array<iNode>, showHeader?: boolean) => {
+        let _currentNodeHeader = ''
         return nodes.map((node, index) => {
-            const nodeName = `${node.name}.${node.namespace} : { portnumber }`;
+            const nodeName = `${node.name}.${node.namespace} : { portnumber }`
+
+            // Only render node kind header when it's the first node or it's a different kind header
+            _currentNodeHeader = index === 0 ? node.kind : _currentNodeHeader !== node.kind ? node.kind : ''
```

Comment:
can we write  like  _currentNodeHeader = (index === 0  || _currentNodeHeader !== node.kind)? node.kind :  '' ?

Reply:
Done